salt/transport/zeromq.py

```diff
@@ -535,10 +536,16 @@ def close(self):
                     self.stream.socket = None
                     self.socket.close()
                 else:
-                    self.stream.close(1)
+                    try:
+                        self.stream.close(1)
+                    except ImportError:
+                        # On teardown of python, this can sometimes throw and ImportError
+                        # because sys.meta_path turns to None
+                        teardown = True
                     self.socket = None
                 self.stream = None
-            if self.context.closed is False:
+            if self.context.closed is False and not teardown:
+                # This hangs if closing the stream causes an import error
                 self.context.term()
```

Comment:
We might have to also catch `ImportError` at this stage too now that we handle it above.

Reply:
At that point, would it be better to wrap the entire clause in a `try/except`?  Starting at line 528.

Reply:
Well, ieadlly yes, but if the last one doesn't throw the exception it will actually terminate the context, while, of it was just one try/except we could potentially not terminate the context.

Were you able to reproduce the issue?

Reply:
Yes, `self.context.term()` hangs when we handle the exception.  I'm not sure whether this is a problem since we are seeing this on teardown only.  Any thoughts?

Reply:
I'd rather see us just remove the `__del__`.